src/macros.rs

```diff
@@ -135,8 +136,9 @@ macro_rules! __log {
     (logger: $logger:expr, target: $target:expr, $lvl:expr, $($arg:tt)+) => ({
         let lvl = $lvl;
         if lvl <= $crate::STATIC_MAX_LEVEL && lvl <= $crate::max_level() {
-            $crate::__private_api::log(
-                $logger,
+            use $crate::__private_api::LogExt;
+
+            $logger.do_log(
```

Comment:
You may need to refer to the trait directly: `LogExt::do_log(...)`, otherwise if the logger has a `do_log` method that is not a trait method, it will be called here.